### PR TITLE
Fixed alignment of the device stack memory

### PR DESCRIPTION
**Pull request by ravil-mobile**

### CODE REVIEW

File: interfaces/cuda/Control.cu [hunk 1]
```diff
@@ -140,6 +140,12 @@ unsigned ConcreteAPI::getGlobMemAlignment() {
   return 128;
 }
 
+size_t ConcreteAPI::alignBytes(size_t requestedBytes) {
```
Comment (Thomas-Ulrich) [inline interfaces/cuda/Control.cu]: Would getSizeAlignedBytes be more precise?

File: interfaces/cuda/Control.cu [hunk 2]
```diff
@@ -140,6 +140,12 @@ unsigned ConcreteAPI::getGlobMemAlignment() {
...
+  auto alignment = getGlobMemAlignment();
+  size_t alignmentFactor = (requestedBytes + alignment - 1) / alignment;
```
Comment (Thomas-Ulrich) [inline interfaces/cuda/Control.cu]: Why the `+ alignment - 1`?

Comment (sebwolf-de) [reply interfaces/cuda/Control.cu]: You search the smallest number which is larger or equal than `requestedBytes` and a multiple of `alignment`. 
So for any `i`, we map `i*alignment+1`, ..., `i*alignment + alignment` to `(i+1) * alignment`. That's what this formula does.

**Review by sebwolf-de — COMMENTED** [review]
Isn't the implementation always the same, so you can move this to the AbstractAPI?

**Comment by ravil-mobile** [comment]
> Isn't the implementation always the same, so you can move this to the AbstractAPI?

I was lazy :smile: . Rework!

File: interfaces/cuda/Control.cu (hunk 2 above)
Comment (Thomas-Ulrich) [reply interfaces/cuda/Control.cu]: ok thx.

File: interfaces/cuda/Control.cu (hunk 1 above)
Comment (ravil-mobile) [reply interfaces/cuda/Control.cu]: I made this function a little bit more generic and gave it a more generic name:
https://github.com/SeisSol/Device/blob/e050438c703a10996c3429d2c0c2e8b583a8af1c/interfaces/common/Common.h#L21-L25

**Comment by ravil-mobile** [comment]
> Isn't the implementation always the same, so you can move this to the AbstractAPI?

done.

**Review by Thomas-Ulrich — APPROVED** [review]

**Pull request merged**